models/frax_finance/bnb/frax_finance_bnb_bribes.sql

```diff
@@ -0,0 +1,213 @@
+with base_date as (
+    SELECT
+      week_array AS week_start,
+      week(date_add('DAY',-4, week_array)) as week,
+      date_add('DAY', 7, week_array) as week_end
+    FROM
+      UNNEST(
+        SEQUENCE(
+          cast('2023-01-05' AS date),
+          current_date,
+          INTERVAL '7' DAY
+        )
+      ) AS t1(week_array)
+      )
+,
+pairs as 
+(
+select 
+distinct
+contract_address,
+output_0 as token0,
+output_1 as token1
+from {{source('frax_finance_bnb','thena_fi_bnb.Pair_call_tokens')}} 
+where (output_0 in (0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40,0x64048A7eEcF3a2F1BA9e144aAc3D7dB6e58F555e) or 
+output_1 in (0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40,0x64048A7eEcF3a2F1BA9e144aAc3D7dB6e58F555e))
+and date(call_block_time) = (
+    select max(date(call_block_time)) from thena_fi_bnb.Pair_call_tokens
+    where (output_0 in (0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40,0x64048A7eEcF3a2F1BA9e144aAc3D7dB6e58F555e) or 
+    output_1 in (0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40,0x64048A7eEcF3a2F1BA9e144aAc3D7dB6e58F555e))
+                            ) 
+and call_success = true
+)
+,
+reserves as (
+select contract_address,
+call_block_time,
+output__reserve0/cast(pow(10,18) as uint256) as reserve0,
+output__reserve1/cast(pow(10,18) as uint256) as reserve1,
+row_number() over(partition by date(call_block_time),contract_address order by call_block_time desc) as rn
+from {{source('frax_finance_bnb','thena_fi_bnb.Pair_call_getReserves')}} 
+where date(call_block_time) >= date('2023-01-04')
+and contract_address in (select distinct contract_address from pairs)
+),
+TVL as (
+select 
+ a.call_block_time as block_time
+,a.contract_address 
+,b.token0 
+,b.token1
+,case when token0 = 0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40 then reserve0
+ when token1 = 0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40 then reserve1
+ end as Frax_reserve
+ ,case when token0 = 0x64048A7eEcF3a2F1BA9e144aAc3D7dB6e58F555e then reserve0
+ when token1 = 0x64048A7eEcF3a2F1BA9e144aAc3D7dB6e58F555e then reserve1
+ end as FrxETH_reserve
+from reserves a
+left join pairs b
+on a.contract_address = b.contract_address 
+where rn = 1
+)
+,
+prices_raw as (
+select *,
+row_number() over(partition by date(hour),contract_address order by hour desc) as rn
+from {{ref('dex_prices')}}
+where blockchain = 'ethereum'
+and contract_address in  (0x5e8422345238f34275888049021821e8e08caa1f,0x853d955aCEf822Db058eb8505911ED77F175b99e, 0x3432B6A60D23Ca0dFCa7761B7ab56459D9C964D0)
```

Comment:
We use strict lower-cased addresses in most chains.

Reply:
I thought that wouldn't matter in varbinary.